### DISCUSSION TITLE
Local SSL https in Local private self hosted environment

### DISCUSSION THREAD

**Discussion by dualizeo** [caprover/caprover#1587] [Q&A]
Greetings! 

I have searched all over the CapRover Discussions and Issues but could not find a clear direction on self hosting Locally private CapRover with https <-

I went through the documentation and it says:
`Side note: You can [install CapRover locally](https://caprover.com/docs/run-locally.html) on your laptop on a private network which is behind NAT (your router). But if you want to enable HTTPS and/or access the apps from outside of your private network, it requires some special setup, like port forwarding.`

I do have knowledge of reverse proxy and I was able to open ports. I was able to successfully download CapRover in my Ubuntu VM machine, but after this step I am not sure how to go about.

Could anyone please provide detail documentation on going about this situation? I am using CloudFlare as a DNS to allow https to reach Let's Encrypt. ( I have knowledge of reverse proxy as I previously setup traefik to get local ssl to work) But not sure how this will work with CapRover.
Thanks for your time!

**Comment by githubsaturn** [comment]
You need to make sure that `http://captain.your.domain.com:80` loads the dashboard correctly. Once that's confirmed, then you can go ahead and just enable the HTTPs on the dashboard.

**Comment by dualizeo** [comment]
Thanks for the reply however I keep getting "1107: Verification Failed" error, 

it seems to be picking up my local IP address from the internal vlan network I made, instead of my outer physical network. As I am running CapRover in a ProxMox Virtual machine environment, and I placed it on the same network of my pfsense Virtual machine ( router / firewall ) so I can separate the VM's on a different network.

So what you are saying is just make a dns entry in my pfsense router to resolve https internally? because it continues to not recognize it.
I did an entry caprovertest.com and placed it to update domain field but same error ( same after opening ports )
I appreciate your responses 🙏

**Reply by githubsaturn** [reply]
You can disable CapRover's internal verification by running this 
https://caprover.com/docs/troubleshooting.html#domain-verification-failed---error-1107

**Reply by dualizeo** [reply]
Thanks got it to work locally! but unfortunately only http and after trying port forwarding, it does not allow me to run **https** after updating my domain. 
So it appears that CapRover does not allow local https for a local environment, it's a shame as there seems to be other users like me interested in this feature, I guess it's time to move on to coolify.

But thanks for your support, hopefully in the future it could be implemented in CapRover v2

**Reply by githubsaturn** [reply]
> So it appears that CapRover does not allow local https for a local environment

This is not true. As long as you have port forward set up on your NAT, you can enable https. What problem did you face?

**Reply by dualizeo** [reply]
Thanks for following up, but after several attempts in updating the domain that points to my cloudflare it kept failing, I even changed my local dns management to use Cloudflare dns but no luck, unless I use my local dns to continue with captain.local for https ? but that did not work. And as I previously mentioned that I am running a vm behind pfsense router vm so that's a different IP.

**Reply by githubsaturn** [reply]
It doesn't matter how many NAS sit between your VM and the public network. As long as you directly point your DNS to the public IP of your router, and you forward port 80 and 443 to go from your router to the VM, everything should work fine. 

Use a non local device (e.g. your cellphone not connected to your wifi) and confirm:
1) http://randomstring.yourdomain.com loads a caprover page (confirming that it is hitting your VM)
2) http://randomstring.yourdomain.com loads a caprover page (after you click on "advance" and bypass the invalid HTTPS check)

After you confirm 1 & 2 => CapRover can work normally.

Make sure when you select your cloudflare setting you use "DNS only". Can you copy a screenshot of your Cloudflare?

**Reply by dualizeo** [reply]
Thanks for not giving up on me, I will try again and report back with results ( I will be testing a different method )
*EDIT* sorry for late response but I went with truenas scale instead as I installed treafik and it doesn't require open ports
